Binding has (was) stopped and the devices are no longer communicating.

**First step**
- [x] I have read the FAQ https://github.com/seime/openhab-esphome?tab=readme-ov-file#faq

**My setup**
- [x] I'm running **OH 5.1.4**
- [x] The ESPHome device is running version **2026.6.2**
- [x] Output of `openhab-cli` command `bundle:list|grep ESPHome` is **4.1.0.202606050534**

**OH Logs**
```
2026-07-05 01:44:34.117 [WARN ] [e.internal.comm.EncryptedFrameHelper] - Error sending message
no.seime.openhab.binding.esphome.internal.comm.ProtocolAPIError: null
	at no.seime.openhab.binding.esphome.internal.comm.EncryptedFrameHelper.encodeFrame(EncryptedFrameHelper.java:246) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.comm.EncryptedFrameHelper.send(EncryptedFrameHelper.java:392) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.handler.ESPHomeHandler.onConnect(ESPHomeHandler.java:400) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.comm.EncryptedFrameHelper.lambda$0(EncryptedFrameHelper.java:209) ~[?:?]
	at com.jano7.executor.KeyRunnable.run(KeyRunnable.java:63) ~[?:?]
	at com.jano7.executor.KeySequentialRunner$KeyRunner.runSafely(KeySequentialRunner.java:102) ~[?:?]
	at com.jano7.executor.KeySequentialRunner$KeyRunner.lambda$run$0(KeySequentialRunner.java:72) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.MonitoredCompositeExecutorService$TimedRunnable.run(MonitoredCompositeExecutorService.java:486) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1583) [?:?]
2026-07-05 01:44:34.119 [WARN ] [e.internal.comm.EncryptedFrameHelper] - Error sending message
no.seime.openhab.binding.esphome.internal.comm.ProtocolAPIError: null
	at no.seime.openhab.binding.esphome.internal.comm.EncryptedFrameHelper.encodeFrame(EncryptedFrameHelper.java:246) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.comm.EncryptedFrameHelper.send(EncryptedFrameHelper.java:392) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.handler.ESPHomeHandler.onConnect(ESPHomeHandler.java:402) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.comm.EncryptedFrameHelper.lambda$0(EncryptedFrameHelper.java:209) ~[?:?]
	at com.jano7.executor.KeyRunnable.run(KeyRunnable.java:63) ~[?:?]
	at com.jano7.executor.KeySequentialRunner$KeyRunner.runSafely(KeySequentialRunner.java:102) ~[?:?]
	at com.jano7.executor.KeySequentialRunner$KeyRunner.lambda$run$0(KeySequentialRunner.java:72) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.MonitoredCompositeExecutorService$TimedRunnable.run(MonitoredCompositeExecutorService.java:486) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1583) [?:?]
2026-07-05 01:44:34.305 [WARN ] [core.thing.internal.ThingManagerImpl] - Failed to normalize configuration for new thing during update 'esphome:device:esphome-soil-moisture-a': {thing/channel=Type description esphome:esphome-soil-moisture-a_latestFirmwareVersion for esphome:device:esphome-soil-moisture-a:latestFirmwareVersion not found, although we checked the presence before.}
2026-07-05 01:45:34.075 [ERROR] [ome.internal.comm.ConnectionSelector] - Error while selecting, stopping thread
java.nio.channels.CancelledKeyException: null
	at sun.nio.ch.SelectionKeyImpl.ensureValid(SelectionKeyImpl.java:75) ~[?:?]
	at sun.nio.ch.SelectionKeyImpl.readyOps(SelectionKeyImpl.java:138) ~[?:?]
	at java.nio.channels.SelectionKey.isReadable(SelectionKey.java:355) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.comm.ConnectionSelector.processKey(ConnectionSelector.java:59) ~[?:?]
	at no.seime.openhab.binding.esphome.internal.comm.ConnectionSelector.lambda$0(ConnectionSelector.java:39) ~[?:?]
	at java.lang.Thread.run(Thread.java:1583) [?:?]
2026-07-05 01:46:18.894 [WARN ] [home.internal.handler.ESPHomeHandler] - [remote-control] Ping responses lacking. Waited 4 times 10s, total of 40s. Last pong received at 2026-07-04T23:45:28.901218058Z. Assuming connection lost and disconnecting
2026-07-05 01:46:18.894 [WARN ] [home.internal.handler.ESPHomeHandler] - [remote-control] Disconnecting. Reason: ESPHome did not respond to ping requests. 4 pings sent with 10 s delay. Will reconnect in 15 seconds
2026-07-05 01:46:18.981 [WARN ] [home.internal.handler.ESPHomeHandler] - [doorbell] Ping responses lacking. Waited 4 times 10s, total of 40s. Last pong received at 2026-07-04T23:45:28.989575909Z. Assuming connection lost and disconnecting
2026-07-05 01:46:18.981 [WARN ] [home.internal.handler.ESPHomeHandler] - [doorbell] Disconnecting. Reason: ESPHome did not respond to ping requests. 4 pings sent with 10 s delay. Will reconnect in 11 seconds

```
**Describe the bug**
The binding has stopped and the devices are no longer communicating. After restarting the binding via the openHAB console, everything works as before again.

--- a/src/main/java/no/seime/openhab/binding/esphome/internal/comm/ConnectionSelector.java
+++ b/src/main/java/no/seime/openhab/binding/esphome/internal/comm/ConnectionSelector.java
@@ -56,8 +56,8 @@ public class ConnectionSelector {
 
     private void processKey(SelectionKey key) {
         EncryptedFrameHelper frameHelper = (EncryptedFrameHelper) key.attachment();
-        logger.trace("Processing key readable={}, connectable={}", key.isReadable(), key.isConnectable());
         try {
+            logger.trace("Processing key readable={}, connectable={}", key.isReadable(), key.isConnectable());
             SocketChannel channel = (SocketChannel) key.channel();
             if (key.isConnectable() && channel.isConnectionPending()) {
                 boolean connected = channel.finishConnect();

